File: Strings/Java/LengthOfTheLastWord.java
/*
LeetCode Question:
https://leetcode.com/problems/length-of-last-word/
*/
public class LengthOfTheLastWord {
    public static int lengthOfLastWord(String s) {
        // Trim any leading or trailing whitespace
        s = s.trim();

        // Find the last space after trimming
        int lastSpaceIndex = s.lastIndexOf(' ');

        // Return the length of the last word
        return s.length() - lastSpaceIndex - 1;
    }

    public static void main(String[] args) {
        System.out.println(lengthOfLastWord("Hello World"));     
    }
}

File: Strings/CPP/LengthOfTheLastWord.cpp
/*
LeetCode Question:
https://leetcode.com/problems/length-of-last-word/
*/
#include <iostream>
#include <string>

using namespace std;

int lengthOfLastWord(string str) {
    // Trim trailing spaces
    while (!str.empty() && str[str.length() - 1] == ' ') {
        str.pop_back();
    }

    int length = 0;
    for (int i = str.length() - 1; i >= 0; --i) {
        if (str[i] == ' ') {
            break;
        }
        length++;
    }

    // Return the length of the last word
    return length;
}

int main() {
    cout << lengthOfLastWord("   Hello World   ") << endl;
    return 0;
}
